interface.go

```diff
@@ -45,20 +50,24 @@ const (
 	ConnStateForwardSecure
 )
 
-// ConnStateCallback is called every time the connection moves to another connection state
-// the callback is called in a new go routine
+// ConnStateCallback is called every time the connection moves to another connection state.
 type ConnStateCallback func(Session, ConnState)
 
-// Config is the configuration for QUIC
+// Config contains all configuration data needed for a QUIC server or client.
 type Config struct {
 	TLSConfig *tls.Config
-	// will be called in a separate goroutine
+	// ConnStateCallback will be called when the QUIC version is successfully negotiated or when the encryption level changes.
+	// If this field is not set, the Dial functions will return only when the connection is forward secure.
+	// The callback may be called in a separate goroutine.
```

Comment:
I don't like that sentence, the "may" sounds like we're allowing the user to do something, where in fact we're reserving us the right to call the callback in a goroutine.
Maybe we could write something along these lines:

> Callbacks have to be thread-safe, since they might be called in a separate goroutine.

or "executed concurrently".